### DISCUSSION TITLE
How to log an event in native android java?

### DISCUSSION THREAD

**Discussion by nschurmann** [invertase/react-native-firebase#4611] [Q&A]
I need to log an event when a user use an android widget, I was trying to import the class `ReactNativeFirebaseAnalyticsModule` but to my surprise, this class doesn't have a public constructor so i can't create an instance nor call the logEvent module. How can I log an event using nothing but java (not javascript) and this library?

**Comment by mikehardy** [comment]
Hi there! There is nothing stopping you from using the firebase-android-sdk APIs directly, it is what I would recommend, there's no need to use the react-native-firebase wrapper APIs at that point - examples are in the upstream quickstart https://github.com/firebase/quickstart-android/tree/master/analytics
